handler/handler.go

```diff
@@ -344,55 +439,34 @@ func writeMessage(w http.ResponseWriter, msg string, status int) {
 	}
 }
 
-// ReplaceAnnotation deletes an annotation for a specific content uuid and adds a new one.
-// It gets the annotations only from UPP skipping V2 annotations because they are not editorially curated.
-func (h *Handler) ReplaceAnnotation(w http.ResponseWriter, r *http.Request) {
-	w.Header().Add("Content-Type", "application/json")
-
-	contentUUID := vestigo.Param(r, "uuid")
-	conceptUUID := vestigo.Param(r, "cuuid")
-
-	tID := tidutils.GetTransactionIDFromRequest(r)
-	ctx := tidutils.TransactionAwareContext(context.Background(), tID)
+func switchToHasBrand(toChange []annotations.Annotation) ([]annotations.Annotation, error) {
+	changed := make([]annotations.Annotation, len(toChange))
+	for idx, ann := range toChange {
+		err := validatePredicate(ann.Predicate)
+		if err != nil {
+			return nil, fmt.Errorf("index %d : %w", idx, err)
+		}
+		if ann.Type == "" {
+			return nil, fmt.Errorf("index %d : annotation missing concept type", idx)
+		}
 
-	writeLog := log.WithField(tidutils.TransactionIDKey, tID).WithField("uuid", contentUUID)
-	oldHash := r.Header.Get(annotations.PreviousDocumentHashHeader)
+		if ann.Predicate == mapper.PredicateIsClassifiedBy && ann.Type == mapper.ConceptTypeBrand {
+			ann.Predicate = mapper.PredicateHasBrand
```

Comment:
Operating on a different slice ensures that we have no side effects if there is an error in the middle of the transformation.

Reply:
Yes, I agree but my question was if actually need no side effects (since at when calling those functions we store the result from them in the same value we are passing to them).

Reply:
I've written it that way more on principle. The overhead is not that big and it makes the code more robust and readable, at least for me.